catalog/async/catalog-async-processingframework/src/main/java/org/codice/ddf/catalog/async/processingframework/impl/InMemoryProcessingFramework.java

```diff
@@ -0,0 +1,294 @@
+                        request = plugin.processUpdate(request);
+                    } catch (PluginExecutionException e) {
+                        LOGGER.debug("Unable to process update request through plugin: {}",
+                                plugin.getClass()
+                                        .getCanonicalName(),
+                                e);
+                    }
+                }
+
+                storeProcessRequest(request);
+                closeInputStream(request);
+            });
+        }
+    }
+
+    @Override
+    public void submitDelete(ProcessRequest<ProcessDeleteItem> input) {
+        if (postProcessPlugins == null || postProcessPlugins.isEmpty()) {
+            LOGGER.debug("postProcessPlugins is empty. Not starting post process thread");
+        } else {
+            threadPool.submit(() -> {
+                ProcessRequest<ProcessDeleteItem> request = input;
+
+                for (PostProcessPlugin plugin : postProcessPlugins) {
+                    try {
+                        request = plugin.processDelete(request);
+                    } catch (PluginExecutionException e) {
+                        LOGGER.debug("Unable to process request through plugin: {}",
+                                plugin.getClass()
+                                        .getCanonicalName(),
+                                e);
+                    }
+                }
+            });
+        }
+    }
+
+    private <T extends ProcessResourceItem> void closeInputStream(ProcessRequest<T> request) {
+        request.getProcessItems()
+                .stream()
+                .map(ProcessResourceItem::getProcessResource)
+                .filter(Objects::nonNull)
+                .forEach(resource -> {
+                    try {
+                        resource.getInputStream()
+                                .close();
+                    } catch (IOException e) {
+                        LOGGER.debug(
+                                "Failed to close stream of asynchronous request. There may be temporary files left in data/tmp.");
+                    }
+                });
+    }
+
+    private <T extends ProcessResourceItem> void storeProcessRequest(
+            ProcessRequest<T> processRequest) {
+        LOGGER.trace("Storing update request post processing change(s)");
+
+        Set<ContentItem> contentItemsToUpdate = new HashSet<>();
+        Set<Metacard> metacardsToUpdate = new HashSet<>();
+
+        for (T item : processRequest.getProcessItems()) {
+            if (item.getProcessResource() == null && item.isMetacardModified()) {
+                metacardsToUpdate.add(item.getMetacard());
+            }
+
+            final ProcessResource processResource = item.getProcessResource();
+            if (processResource != null && processResource.isModified()) {
+                try {
+                    byte[] byteArray = IOUtils.toByteArray(processResource.getInputStream());
```

Comment:
I am worried that this will cause OOM exceptions for large resources.  Can we copy this to a `FileBackedOutputStream` and use that to get a `ByteSource`?

@codice/security